### PR TITLE
added defcomp info for QuantinuumBackend

### PR DESCRIPTION
**Pull request by CalMacCQ**
Added docs page for default compilation for the Quantinuum Backend. Once I have the right content it should be quick to do for the other backends.

### CODE REVIEW

File: modules/pytket-quantinuum/docs/intro.txt [hunk 1]
```diff
@@ -17,6 +17,87 @@ and Windows. To install, run:
...
+
+
+.. list-table:: **Default compilation pass for the Quantinuum Backend**
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * - optimisation_level = 0
+     - optimisation_level = 1 [1]
+     - optimisation_level = 2
+   * - DecomposeClassicalExp
```
Comment (ss2165) [inline modules/pytket-quantinuum/docs/intro.txt:45]: note this will need changing after #385

File: modules/pytket-quantinuum/docs/intro.txt [hunk 2]
```diff
@@ -17,6 +17,87 @@ and Windows. To install, run:
...
+     - DecomposeClassicalExp
+     - DecomposeClassicalExp
+   * - DecomposeBoxes
+     - DecomposeBoxes
+     - DecomposeBoxes
+   * - self.rebase_pass
+     - SynthesiseTket
+     - FullPeepholeOptimise
+   * -
+     - self.rebase_pass
```
Comment (ss2165) [inline modules/pytket-quantinuum/docs/intro.txt:55]: add a footnote explaining what this is

File: modules/pytket-quantinuum/docs/intro.txt [hunk 3]
```diff
@@ -17,6 +17,87 @@ and Windows. To install, run:
...
+     - SimplifyInitial [3]
+
+* [1] If no value is specified then ``optimisation_level`` defaults to a value of 1.
+
+* [2] ``auto_squash_pass`` has arguments ``auto_squash_pass(OpType.PhasedX,OpType.Rz)``
+
+* [3] ``SimplifyInitial`` has arguments  ``SimplifyInitial(allow_classical=False, create_all_qubits=True, xcirc=xcirc)``
+
+.. note::   If ``optimisation_level = 0`` the device constraints are solved but no additional optimisation is applied. Setting ``optimisation_level = 1`` applies some light optimisations to the circuit equivalent to the
+            ``get_compiled_circuit`` method. More intensive optimisation is applied by level 2 at the expense of increased runtime.
```
Comment (ss2165) [inline modules/pytket-quantinuum/docs/intro.txt:74]: you're talking about the `get_compiled_circuit` method, so "equivalent" doesn't make sense here?

File: modules/pytket-quantinuum/docs/intro.txt [hunk 4]
```diff
@@ -17,6 +17,87 @@ and Windows. To install, run:
...
+   * -
+     - SimplifyInitial [3]
+     - auto_squash_pass [2]
+   * -
+     -
+     - SimplifyInitial [3]
+
+* [1] If no value is specified then ``optimisation_level`` defaults to a value of 1.
+
+* [2] ``auto_squash_pass`` has arguments ``auto_squash_pass(OpType.PhasedX,OpType.Rz)``
```
Comment (ss2165) [inline modules/pytket-quantinuum/docs/intro.txt:69]:
```suggestion
* [2] ``auto_squash_pass`` has arguments ``auto_squash_pass({OpType.PhasedX,OpType.Rz})``
```

**Review by ss2165 — CHANGES_REQUESTED** [review]
Almost there! Just a couple of requests:

* login, logout,  device_state, cancel missing from the additional functionality section
* the status, result handle type and result type sections could do with their own headings

File: modules/pytket-quantinuum/docs/intro.txt [hunk 5]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
 .. note::   Requires a `Quantinuum <https://www.quantinuum.com/>`_ account.
             Use ``QuantinuumBackend.login()`` to log in using your email and password.
 
+**Devices**
+
+* ``H1-1``, ``H1-2``: Quantum Computers, submit specifically to ``H1-1`` or ``H1-2`` by using the device name
+* ``H1-1E``, ``H1-2E``: Device-specific Emulators of ``H1-1`` and ``H1-2``. These emulators run remotely on servers and require credentials.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:23]:
```suggestion
* ``H1-1E``, ``H1-2E``: Device-specific emulators of ``H1-1`` and ``H1-2``. These emulators run remotely on servers and require credentials.
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 6]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+* ``H1-1SC``, ``H1-2SC``: Device-specific Syntax Checkers, check compilation of a quantum circuit against device-specific instructions, returns completed if syntax is correct along with the H-Series Quantum Credits (HQCs), returns failed if the syntax is incorrect along with the error.
+* ``H1``: Submit to Quantinuum System Model ``H1`` machines, job is run on ``H1-1`` or ``H1-2`` based on soonest available device.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:25]:
```suggestion
* ``H1``: Submit to Quantinuum System Model ``H1`` machines: job is run on ``H1-1`` or ``H1-2`` based on soonest available device.
```
Is "Quantinuum System Model" a proper name?

File: modules/pytket-quantinuum/docs/intro.txt [hunk 7]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+The H-series devices and emulators produce shots based results and therefore require measurements. The emulators use noise models based on the real devices.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:27]:
```suggestion
The H-series devices and emulators produce shots-based results and therefore require measurements. The emulators use noise models based on the real devices. 
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 8]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+The H-series devices and emulators produce shots based results and therefore require measurements. The emulators use noise models based on the real devices. 
+
+**Job Statuses**
+
+When using the Quantinuum ``Backend`` to run circuits there are several possible circuit statuses.
+
+* queued - The job has been queued but has not yet been run.
+* running - circuit is currently being run on the device/emulator.
+* completed - the job has finished.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:35]:
```suggestion
* completed - The job has finished.
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 9]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+* failed - job has failed.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:36]:
```suggestion
* failed - The job has failed.
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 10]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+* cancelling - the job is in the process of being cancelled.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:37]:
```suggestion
* cancelling - The job is in the process of being cancelled.
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 11]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+* cancelled - the job has been cancelled.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:38]:
```suggestion
* cancelled - The job has been cancelled.
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 12]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+**Default Compilation**
+
+Every ``Backend`` in pytket has its own ``default_compilation_pass`` method. 
+This method applies a sequence of optimisations to a circuit depending on the value of an ``optimisation_level`` parameter. 
+This default compilation will ensure that the circuit meets all the constraints required to run on the ``Backend``. 
+The passes applied by different levels of optimisation are specified in the table below.
+
+
+.. list-table:: **Default compilation pass for the Quantinuum Backend**
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:51]:
```suggestion
.. list-table:: **Default compilation pass for the Quantinuum backends**
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 13]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+   * -
+     - `SimplifyInitial [4] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.SimplifyInitial>`_
+     - `auto_squash_pass [3] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.auto_rebase.auto_squash_pass>`_
+   * -
+     -
+     - `SimplifyInitial [4] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.SimplifyInitial>`_
+
+* [1] If no value is specified then ``optimisation_level`` defaults to a value of 1.
+
+* [2] ``self.rebase_pass`` is a rebase that converts the circuit to the Quantinuum native gate set {Rz, PhasedX, ZZMax, ZZPhase}.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:79]:
```suggestion
* [2] ``self.rebase_pass`` is a rebase that converts the circuit to the Quantinuum native gate set (e.g. {Rz, PhasedX, ZZMax, ZZPhase}).
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 14]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+* [3] ``auto_squash_pass`` has arguments ``auto_squash_pass({OpType.PhasedX,OpType.Rz})``
+
+* [4] ``SimplifyInitial`` has arguments  ``SimplifyInitial(allow_classical=False, create_all_qubits=True, xcirc=xcirc)``
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:83]: This is incomplete without code context (what is `xcirc`?). Do we wantthis level of detail here?

File: modules/pytket-quantinuum/docs/intro.txt [hunk 15]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+.. note::   If ``optimisation_level = 0`` the device constraints are solved but no additional optimisation is applied. Setting ``optimisation_level = 1`` applies some light optimisations to the circuit. More intensive optimisation is applied by level 2 at the expense of increased runtime.
+
+**Device Predicates**
+
+Circuits must satisfy the following predicates in order to run on the ``QuantinuumBackend``.
+
+* `NoSymbolsPredicate  <https://cqcl.github.io/tket/pytket/api/predicates.html#pytket.predicates.NoSymbolsPredicate>`_  - Parameterised gates must have numerical parameters when the circuit is executed.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:91]:
```suggestion
* `NoSymbolsPredicate  <https://cqcl.github.io/tket/pytket/api/predicates.html#pytket.predicates.NoSymbolsPredicate>`_  : Parameterised gates must have numerical parameters when the circuit is executed.
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 16]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+* `GateSetPredicate  <https://cqcl.github.io/tket/pytket/api/predicates.html#pytket.predicates.GateSetPredicate>`_ : To view supported Ops run ``QuantinuumBackend.backend_info.gate_set``
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:92]:
```suggestion
* `GateSetPredicate  <https://cqcl.github.io/tket/pytket/api/predicates.html#pytket.predicates.GateSetPredicate>`_ : To view supported Ops run ``QuantinuumBackend.backend_info.gate_set``.
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 17]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+* `MaxNQubitsPredicate(12) <https://cqcl.github.io/tket/pytket/api/predicates.html#pytket.predicates.MaxNQubitsPredicate>`_ for ``H1-1``, ``H1-2`` and ``H1-2E``. ``MaxNQubitsPredicate(20)`` for ``H1-1E``.
+
+
+**Additional Backend Capabilities**
+
+The Quantinuum ``Backend`` has a ``cost`` method. This calculates the cost (in H-series Quantum Credits) required to execute the circuit for the specified number of shots.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:98]:
```suggestion
The Quantinuum ``Backend`` has a ``cost`` method. This calculates the cost (in HQS) required to execute the circuit for the specified number of shots.  
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 18]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+The ``process_circuits`` method for the QuantinuumBackend accepts the following additional keyword arguments.
+
+*  ``postprocess`` : boolean flag to allow classical postprocessing.
+*  ``noisy_simulation`` : boolean flag to specify whether the simulator should
+   perform noisy simulation with an error model (default value is ``True``).
+*  ``group`` : string identifier of a collection of jobs, can be used for usage tracking.
+*  ``max_batch_cost``: maximum HQC usable by submitted batch, default is 500.
+*  ``batch_id`` : first ``jobid`` of the batch to which this batch of circuits should be submitted. Job IDs can be retrieved from ResultHandle using ``backend.get_jobid(handle)``.
+*  ``close_batch`` : boolean flag to close the batch after the last circuit in the job, ``default=True``.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:111]:
```suggestion
*  ``close_batch`` : boolean flag to close the batch after the last circuit in the job (default value is ``True``).
```

File: modules/pytket-quantinuum/docs/intro.txt [hunk 19]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+For the Quantinuum ``Backend``, ``process_circuits`` returns a ResultHandle object containing a ``job_id`` and a postprocessing ( ``ppcirc``) circuit if there is one. The ``logout()`` method clears stored JSON web tokens and will have to sign in to access the Quantinuum API.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:113]:
```suggestion
For the Quantinuum ``Backend``, ``process_circuits`` returns a ``ResultHandle`` object containing a ``job_id`` and a postprocessing ( ``ppcirc``) circuit if there is one. The ``logout()`` method clears stored JSON web tokens and will have to sign in to access the Quantinuum API.
```

Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:113]: Are there some words missing before "will have to sign in", e.g. "after calling this method the user" ...?

File: modules/pytket-quantinuum/docs/intro.txt [hunk 20]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+Every Quantinuum ``Backend`` supports mid circuit measurements.
```
Comment (cqc-alec) [inline modules/pytket-quantinuum/docs/intro.txt:115]:
```suggestion
Every Quantinuum ``Backend`` supports mid-circuit measurements.
```

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: modules/pytket-quantinuum/docs/intro.txt [hunk 21]
```diff
@@ -17,6 +17,104 @@ and Windows. To install, run:
...
+
+
 .. toctree::
     api.rst
-    changelog.rst
+    changelog.rst
```
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply modules/pytket-quantinuum/docs/intro.txt:120]: I'm not sure what happened here was the `changelog.rst` line not showing?

**Review by ss2165 — CHANGES_REQUESTED** [review]

File: modules/pytket-quantinuum/docs/intro.txt [hunk 22]
```diff
@@ -11,12 +11,120 @@ be executed on Quantinuum's quantum devices.
...
+    pip install pytket-quantinuum
 
-.. note::   Requires a `Quantinuum <https://www.quantinuum.com/>`_ account.
-            Use ``QuantinuumBackend.login()`` to log in using your email and password.
+.. note::   Using the ``QuantinuumBackend`` requires a `Quantinuum <https://www.quantinuum.com/>`_ account.
+            The user will be prompted for their login credentials when making API calls such as calling the ``process_circuits`` method or querying ``backend_info`` .
+
+Devices
+=======
+The pytket-quantinuum extension allows the user to access the following quantum devices, emulators and syntax checkers. These backends can be initialised  by passing the device name as a string to the ``QuantinuumBackend`` class.
```
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply modules/pytket-quantinuum/docs/intro.txt]: Is what I've added on this adequate now?

**Review by ss2165 — DISMISSED** [review]

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**